org.lflang/src/org/lflang/generator/c/CGenerator.xtend

```diff
@@ -378,8 +378,8 @@ class CGenerator extends GeneratorBase {
                 if (action.origin == ActionOrigin.PHYSICAL) {
                     // If the unthreaded runtime is requested, use the threaded runtime instead
                     // because it is the only one currently capable of handling asynchronous events.
-                    if (targetConfig.threads < 1) {
-                        targetConfig.threads = 1
+                    if (!targetConfig.threading) {
+                        targetConfig.threading = true
```

Comment:
This is meaning-preserving, but I'm not sure automatically enabling the multi-threaded library works in all scenarios (particularly bare-iron ones).